core/app/[locale]/(default)/account/[tab]/page-data.ts

```diff
@@ -0,0 +1,111 @@
+import { cache } from 'react';
+
+import { getSessionCustomerId } from '~/auth';
+import { client } from '~/client';
+import { FORM_FIELDS_FRAGMENT } from '~/client/fragments/form-fields';
+import { graphql, VariablesOf } from '~/client/graphql';
+import { revalidate } from '~/client/revalidate-target';
+
+const CustomerSettingsQuery = graphql(
```

Comment:
This is kind of hard to explain but I feel `settings-content` and `addresses-content` should both have a gql fragment of the info they require, and this query should just be a recompilation of both fragments, and you simply pass data to both components and if typed correctly they should just have type of the fragment data.

Similar to: https://github.com/bigcommerce/catalyst/blob/main/core/app/%5Blocale%5D/(default)/login/page.tsx#L20

Reply:
Sounds good to me